api/src/main/java/io/grpc/ServerRegistry.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Registry of {@link ServerProvider}s. The {@link #getDefaultRegistry default instance} loads
```

Comment:
Should we also have an `XdsServerProvider` and be part of this registry? Without that  `Grpc.newServerBuilderForPort` cannot be used to build an XdsServer right?

Reply:
No, we wouldn't want that. XdsServer behaves dramatically differently that other servers and the user should opt-in to its behavior. Just because you add a dependency on grpc-xds doesn't mean you should be using xDS load balancing in Channels and the xds integration in XdsServer.

Reply:
ok

Reply:
ok